app/models/service_template_transformation_plan/validate_config_info.rb

```diff
@@ -9,31 +9,94 @@ module ClassMethods
     def validate_config_info(options)
       config_info = options[:config_info]
 
+      # ------------------------------------------------------------------------
+      # TransformationMapping validation
+      # ------------------------------------------------------------------------
       mapping = if config_info[:transformation_mapping_id]
                   TransformationMapping.find(config_info[:transformation_mapping_id])
                 else
                   config_info[:transformation_mapping]
                 end
-
+      # mapping.invalid? added as a part of validating migration plan
       raise _('Must provide an existing transformation mapping') if mapping.blank?
+      unless mapping.valid?
+        raise _('Transformation mapping was invalid.  ' + mapping.errors.messages)
+      end
 
+      # ------------------------------------------------------------------------
+      # check for openstack as a target for Flavors and SecurityGroup validation
+      # ------------------------------------------------------------------------
+      mapping_item        = mapping.transformation_mapping_items.find_by(:destination_type => "CloudTenant")
+      cloud_tenant        = mapping_item.destination if mapping_item.present?
+      osp_flavors         = cloud_tenant.flavors if cloud_tenant.present?
+      osp_security_groups = cloud_tenant.security_groups if cloud_tenant.present?
```

Comment:
It would flow a little better if we set these down by where we are validating them https://github.com/ManageIQ/manageiq/pull/19283/files#diff-a3f646e512ab13f908b88561f1e5ca2dR77